flash/image/embedding/vissl/hooks.py

```diff
@@ -81,6 +82,10 @@ def on_start(self, task: "flash.image.embedding.vissl.adapter.MockVISSLTask") ->
 
         task.loss.info_criterion.precompute_pos_neg_mask()
 
+        # Re-register params / devices
+        _recursive_register(task.loss)
+        task.loss.to(task.vissl_adapter.adapter_task.device, task.vissl_adapter.adapter_task.dtype)
```

Comment:
Do we need to register again? Wouldn't just `task.loss.to(<device>, <dtype>)` work?

Checked locally: tests pass if we remove `_recursive_register` here. But I maybe wrong. :)

Reply:
Removed :smiley: